src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -50,13 +77,11 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
                 if (last is null)
                 {
                     // Deconstructions with no effects lower to nothing. For example, `(_, _) = (1, 2);`
-                    result = null;
-                    temps.Free();
-                    _ = effects.ToImmutableAndFree();
+                    return null;
                 }
                 else
                 {
-                    result = _factory.Sequence(temps.ToImmutableAndFree(), effects.ToImmutableAndFree(), last);
+                    return _factory.Sequence(temps.ToImmutable(), effects.ToImmutableAndClear(), last);
```

Comment:
>ToImmutable [](start = 51, length = 11)

Should the `temps` also be cleared (like you did for the `effects`)?

Reply:
temps is also initialized in `GetAssignmentTargetsAndSideEffects`, if we clear we discard those as well, but here we possibly passing more locals than needed.

Still trying to wrap my head around pool management over all effects and temps.